infrastructure/assumptions/resources/fetch-and-post-result.html

```diff
@@ -0,0 +1,8 @@
+<script>
+  window.addEventListener("message", function (event) {
+    fetch(event.data)
+        .then(response => {parent.postMessage(response.ok, "*")})
+        .catch(error => {parent.postMessage(error.toString(), "*")});
```

Comment:
In order to assert only that the error was a "TypeError", which is (roughly) what `assert_throws_js` would do, I'd suggest instead posting `error.name` here.

And to make the failure more obvious in the case the fetch succeeds, I'd suggest a more descriptive string. All in all:

```suggestion
        .then(response => {parent.postMessage(`fetch succeeded with status ${response.status}`, "*")})
        .catch(error => {parent.postMessage(`fetch failed with ${error.name}`, "*")});
```

It could be made a bit less verbose, but something like that.

Reply:
Great idea, done.